apps/event_bus/lib/event_bus.ex

```diff
@@ -23,23 +25,23 @@ defmodule EventBus do
     end
   end
 
-  def cast_event(:order_cancelled, %EventBus.OrderCancelled{} = payload),
-    do: dispatch_event(:order_cancelled, payload)
+  def cast_event(:order_cancelled, payload),
+    do: dispatch_event(:order_cancelled, %EventBus.OrderCancelled{order: payload})
```

Comment:
we need to drop the references to the struct %EventBus from the EventBus app as it  will be removed at  some point to leave the exchange completely isolated.

One of the things we'll eventually do is create a Mock EventBus that will ship with the exchange app and is the default configured Adapter. 

We can also provide future different adapters  that will ship with the exchange and users of the exchange either use the provided ones or config the exchange to use custom